$MirOS$

This patch makes the following work more smooth,
because the 'tail -f' only lets mksh read the
second-to-last line (depends on "daemon ovpn")

|	#!/bin/mksh
|	export PATH=/sbin:/bin:/usr/bin:/usr/sbin:/usr/mpkg/sbin
|
|	function waitforit
|	{
|		tail -f /var/log/messages | while read line; do
|			[[ $line = \
|			    *@( ovpn\[)*@(: Initialization Sequence Completed)* ]] \
|			    && break
|		done
|	}
|
|	function runvpn
|	{
|		waitforit &
|		openvpn --config ovc.cfg $* || exit 1
|		wait
|	}
|
|	cd $(dirname "$0")
|	if [[ $1 = @(r|R)* ]]; then
|		runvpn --redirect-gateway def1
|	else
|		runvpn
|	fi
|	# insert post-init stuff here
|	exit 0

--- init.c.orig	Wed Apr  5 06:42:09 2006
+++ init.c	Thu Jul 27 11:46:39 2006
@@ -594,6 +594,7 @@ initialization_sequence_completed (struc
 #endif
   else
     msg (M_INFO, "%s", message);
+  msg (M_INFO, "%s", "done.");
 
   /* Flag remote_list that we initialized */
   if ((flags & (ISC_ERRORS|ISC_SERVER)) == 0 && c->c1.remote_list && c->c1.remote_list->len > 1)
